lib/msf/ui/console/module_option_tab_completion.rb

```diff
@@ -0,0 +1,350 @@
+            refname
+          end
+          @cache_payloads
+        end
+
+        #
+        # Provide valid session options for the current post-exploit module
+        #
+        def option_values_sessions
+          if active_module.respond_to?(:compatible_sessions)
+            active_module.compatible_sessions.map { |sid| sid.to_s }
+          end
+        end
+
+        #
+        # Provide valid target options for the current exploit
+        #
+        def option_values_targets
+          res = []
+          if (active_module.targets)
+            1.upto(active_module.targets.length) { |i| res << (i-1).to_s }
+            res += active_module.targets.map(&:name)
+          end
+          return res
+        end
+
+        #
+        # Provide valid action options for the current module
+        #
+        def option_values_actions
+          res = []
+          if (active_module.actions)
+            active_module.actions.each { |i| res << i.name }
+          end
+          return res
+        end
+
+        #
+        # Provide valid nops options for the current exploit
+        #
+        def option_values_nops
+          framework.nops.map { |refname, mod| refname }
+        end
+
+        #
+        # Provide valid encoders options for the current exploit or payload
+        #
+        def option_values_encoders
+          framework.encoders.map { |refname, mod| refname }
+        end
+
+        #
+        # Provide the target addresses
+        #
+        def option_values_target_addrs
+          res = [ ]
+          res << Rex::Socket.source_address()
+          return res if not framework.db.active
+          # List only those hosts with matching open ports?
+          mport = self.active_module.datastore['RPORT']
+          if mport
+            mport = mport.to_i
+            hosts = {}
+            framework.db.services.each do |service|
+              if service.port == mport
+                hosts[ service.host.address ] = true
+              end
+            end
+            hosts.keys.each do |host|
+              res << host
+            end
+            # List all hosts in the database
+          else
+            framework.db.hosts.each do |host|
+              res << host.address
+            end
+          end
+          return res
+        end
+
+        #
+        # Provide the target ports
+        #
+        def option_values_target_ports
+          res = [ ]
+          return res if not framework.db.active
+          return res if not self.active_module.datastore['RHOST']
+          host = framework.db.has_host?(framework.db.workspace, self.active_module.datastore['RHOST'])
+          return res if not host
+          framework.db.services.each do |service|
+            if service.host_id == host.id
+              res << service.port.to_s
+            end
+          end
+          return res
+        end
+      end
+    end
+  end
+end
```

Comment:
Looks like you're missing a trailing new line here 👀 

It's best practice to add a new line to the end of your files, you can do this manually - or you can modify your rubymine settings to do it for you automatically:

![image](https://user-images.githubusercontent.com/60357436/95485918-43a5ac80-098a-11eb-8c52-326dfd2b3e3a.png)

Reply:
Context:

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
Ok, will get those changes made to my settings and try get this fixed.